hack/violation_exceptions.list

```diff
@@ -0,0 +1,14 @@
+API rule violation: names_match,github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2,AlibiExplainerSpec,StorageURI
```

Comment:
How is this file generated ?

Reply:
Just the same as in the script:
```
go run vendor/k8s.io/kube-openapi/cmd/openapi-gen/openapi-gen.go \
    --input-dirs github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2,knative.dev/pkg/apis \
    --output-package github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2/ --go-header-file hack/boilerplate.go.txt \
    -r $CURRENT_VIOLATION_EXCEPTIONS
```
The -r option will generate the report, the one checked in is the current one and later if user changed something mattered here it will be different.
Notice for the openapi itself it did not treat this as error, the return code is still 0 and everything is generated as is even there's report saying it's something not obey to convention.

Reply:
Thanks! Can we fix these warnings? kube-openapi name validation rules expect the field name in Go code and field name in json to be exactly same with just the first letter in different case (Go code requires CamelCase, json requires camelCase).

Reply:
Yes we can fix it that way. But as I mentioned these are all basic data structure and there will be too too much places to change. I think it's too risky so suggest we can just follow the way of kubernetes which is to keep the existing code, but for additional one in future will report error to require either fix the naming violation or add new item to the exception file(not recommended).